### PR TITLE
Remove manual memory management in `GdbConnection::reply_get_thread_list()`

### PR DESCRIPTION
**Pull request by mkmodrzew**
I would want to propose part of the solution for https://github.com/mozilla/rr/issues/820 issue, where I replaced manually memory management for combining a packet string with `std::stringstream`.

Moreover I found little bug in previous implementation: `maxlen` variable didn't counted dot sign from `"p%02x.%02x,"` in `snprintf`, therefor value was lesser than it probably should be.

### CODE REVIEW

File: src/GdbConnection.cc [hunk 1]
```diff
@@ -1646,31 +1647,32 @@ void GdbConnection::reply_get_stop_reason(GdbThreadId which, int sig) {
...
   } else {
-    ssize_t maxlen =
-        1 /*m char*/ +
-        threads.size() * (1 /*p*/ + 2 * sizeof(threads[0]) + 1 /*,*/) +
-        1 /*\0*/;
-    char* str = (char*)xmalloc(maxlen);
-    int offset = 0;
-
-    str[offset++] = 'm';
+    std::stringstream sstr;
```
[the remark this answers is not in the document]
Comment (mkmodrzew) [reply src/GdbConnection.cc]: Fixed

File: src/GdbConnection.cc [hunk 2]
```diff
@@ -1646,31 +1647,32 @@ void GdbConnection::reply_get_stop_reason(GdbThreadId which, int sig) {
...
+    str.back() = 0;
 
-    write_packet(str);
-    free(str);
+    ssize_t maxlen =
+        1 /*m char*/ +
+        threads.size() * (1 /*p*/ + 2 * sizeof(threads[0]) + 1 /*,*/);
+    if (maxlen < str.length()) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (mkmodrzew) [reply src/GdbConnection.cc]: Fixed

**Comment by mkmodrzew** [comment]
Fixed

**Comment by rocallahan** [comment]
Thanks!

**Pull request merged**